service/history/api/recordactivitytaskstarted/api.go

```diff
@@ -145,6 +146,7 @@ func recordActivityTaskStarted(
 	response := &historyservice.RecordActivityTaskStartedResponse{
 		ScheduledEvent:              scheduledEvent,
 		CurrentAttemptScheduledTime: ai.ScheduledTime,
+		Priority:                    priorities.Merge(mutableState.GetExecutionInfo().Priority, ai.Priority),
```

Comment:
This is interesting. So, there's technically a possibility that this may not be the same as the priority used to dispatch the task, since that one is calculated at schedule time and this is at start time. (Suppose the workflow was mutated between schedule and start.)

This priority is just for downstream uses, so I think actually this is better, the mutation takes effect immediately. The alternative would be storing the "scheduled priority" in activity info (recalculated on each retry), but I don't really see the point in doing that.

Reply:
Right; I took your suggestion from our convo as "bind the priority as late as possible".